### DISCUSSION TITLE
On boundary conditions

### DISCUSSION THREAD

**Discussion by chairmanmao256** [mdolab/dafoam#277] [General]
Hi! I've noticed that the boundary condition interface provided in `runScript.py` can only support a few boundary conditions (`inout`, `zeroGradient`, and so on), but I can still set other boundary conditions (such as `totalPressure`) in the `0.orig` directory and get DAFoam run without throwing out an error. So I would like to ask if DAFoam supports all the boundary conditions provided by OpenFOAM., regardless of whether a boundary condition can be set in the interface of `runScript.py`.

**Comment by friedenhe** [comment]
Unfortunately, the totalPressure is not supported. We tried that boundary condition but the adjoint accuracy was not accurate. Actually, any boundary condition that depends on more than two state variables, e.g., U, p, etc. has problems... We haven't gotten a chance to fix this issue yet...

**Reply by chairmanmao256** [reply]
Got it! Thank you!
